include/zephyr/arch/riscv/irq.h

```diff
@@ -29,14 +29,70 @@ extern void arch_irq_enable(unsigned int irq);
 extern void arch_irq_disable(unsigned int irq);
 extern int arch_irq_is_enabled(unsigned int irq);
 
-#if defined(CONFIG_RISCV_HAS_PLIC) || defined(CONFIG_RISCV_HAS_CLIC)
+#if defined(CONFIG_RISCV_HAS_PLIC) || defined(CONFIG_RISCV_HAS_CLIC) || \
+	defined(CONFIG_RISCV_CUSTOM_INTERRUPT_CONTROLLER)
 extern void z_riscv_irq_priority_set(unsigned int irq,
 				     unsigned int prio,
 				     uint32_t flags);
 #else
 #define z_riscv_irq_priority_set(i, p, f) /* Nothing */
 #endif /* CONFIG_RISCV_HAS_PLIC || CONFIG_RISCV_HAS_CLIC */
 
+#if defined(CONFIG_RISCV_CUSTOM_INTERRUPT_CONTROLLER)
+
+/**
+ * @brief Enable an interrupt line for RISC-V custom interrupt controller.
+ *
+ * This routine enables an interrupt line for RISC-V custom interrupt controller.
+ * Should be implemented by custom interrupt controller.
+ *
+ * @param irq IRQ number to enable
+ */
+void z_irq_enable(unsigned int irq);
```

Comment:
@nashif I'm out of touch with this topic. What's the latest on the `z_*` prefix?

Reply:
Hi @nashif , 

Is it fine if we use z_* prefix for RISC-V wide API for custom interrupt controllers? 
Such API is to be instantiated by the particular RISC-V custom interrupt controller driver in /drivers/interrupt_controller.
Additionally we consider the special case when the interrupt_controller is soc-specific and the driver is placed in /soc/riscv

What do you think?

Reply:
Hi @nashif ,

There was no response from you unfortunately

Reply:
Hi @carlocaione , @nashif , @fkokosinski , I came to the Dev Meeting yesterday, but unfortunately you were not participating, so we weren't able to discuss the API. How would we proceed? Would you come to the next Dev Meeting, or maybe let's discuss it on other meeting, like Architecture WG? Or maybe you could suggest your proposal for this API here? This goes terribly slow

Reply:
Hi @nashif ,

Thanks for your reply! I've checked your proposal and I think I've got your idea. 
Please correct me if I'm wrong regarding the next steps here:

1) Instead of z_irq_* functions introduced below we should bring the new prefix, platfrom_*, and change the API like this:

```
void platform_irq_init(void);
void platform_irq_enable(unsigned int irq);
void platform_irq_disable(unsigned int irq);
int platform_irq_is_enabled(unsigned int irq);
```

2) We still want to define this API on include/zephyr/arch/riscv/ level and not e.g. include/zephyr/arch/ level.

3) We should change RISCV_CUSTOM_INTERRUPT_CONTROLLER to more generic PLATFORM_HAS_CUSTOM_INTERRUPT_CONTROLLER and move this config from arch/riscv/Kconfig to arch/Kconfig

Reply:
Will push an update version of #61547 but while working on it, I realized that this issue and can be dealt with by providing an interrupt controller driver and should be a much cleaner solution than doing this using overrides and custom hooks, this is what drivers (and especially intc driver are supposed to do). So why can't this be solved using a driver? all of the above calls are basically what intc drivers are supposed to provide anyways. I will push my changes regardless, but we might to rethink the interrupt related part in it.

Reply:
Hi @nashif , 

> all of the above calls are basically what intc drivers are supposed to provide anyways

Absolutely! Please take a look at how all those arch_irq_xxx() functions are implemented for RISC-V, it is just one or some other form of redirection to particular unique intc_ function.
And those functions are indeed should be implemented by any IC, no matter standard or custom one.
So introduction of the clean API mandatory for all the IC implementation may remove this level of indirection entirely.

We still have ARCH-specifics, but it does look, like this specifics could be kept on arch/ level instead of sipping to soc/ level (like we have with RISC-V) just for the sake of calling the unique intc_ function within a particular driver

Reply:
#66505 could become a generic solution